### PR TITLE
Get version number from tag using versioningit

### LINKED ISSUE #150: Get version number from git tag
Use [versioningit](https://versioningit.readthedocs.io/) to create the version number automatically based on the tags in the repository. This change will add `versioningit` as a new build dependency.

I will work on this.

### PR DESCRIPTION
**Pull request by backmari**
Resolves #150. This change eliminates the need for a hard coded version number, since the package version number will be generated from the git tag.

### CODE REVIEW

**Comment by bmaranville** [comment]
Can you describe how we would use this?  How do we update the version number?

**Comment by backmari** [comment]
@bmaranville Good question. Is there a good place where I could add documentation?

The version number will be generated based on the last git tag, currently it is showing `0.9.4.dev656`, which means 656 commits on top of the last tag `0.9.3`:

```
~/projects/refl1d_development/bumps_fork/bumps$ versioningit  # print the current version
0.9.4.dev656
~/projects/refl1d_development/bumps_fork/bumps$ pip install .
Processing /home/u5z/projects/refl1d_development/bumps_fork/bumps
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: typing-extensions>=3.7.4 in /home/u5z/mambaforge/envs/bumps/lib/python3.12/site-packages (from bumps==0.9.4.dev656) (4.12.2)
Building wheels for collected packages: bumps
  Building wheel for bumps (pyproject.toml) ... done
  Created wheel for bumps: filename=bumps-0.9.4.dev656-py3-none-any.whl size=424103 sha256=d2df2362543f42734b0e3a537b75b3840a750ba45c4a9818011cea6b47a2ba51
  Stored in directory: /tmp/pip-ephem-wheel-cache-clw2gipo/wheels/12/be/91/cc66fbdf868b50988353f2d6e09de408a21527c4f1d3e7c3f9
Successfully built bumps
Installing collected packages: bumps
  Attempting uninstall: bumps
    Found existing installation: bumps 0.9.4.dev655+d202409101545
    Uninstalling bumps-0.9.4.dev655+d202409101545:
      Successfully uninstalled bumps-0.9.4.dev655+d202409101545
Successfully installed bumps-0.9.4.dev656
```

File: doc/developer/index.rst
Comment (bmaranville) [inline doc/developer/index.rst:1]: Can we merge this file with doc/getting_started/contributing.rst?

File: pyproject.toml
Comment (bmaranville) [inline pyproject.toml:1]: I hate to go back and forth on this, but if in the developer docs we recommend running versioningit from the command line, we should definitely install it both in the `dev` dependencies and in `build-system.requires`

**Review by bmaranville — CHANGES_REQUESTED** [review]

File: pyproject.toml
Comment (backmari) [reply pyproject.toml:1]: good point, I will add it to the `dev` dependencies

**Review by bmaranville — APPROVED** [review]

**Pull request merged**